### PR TITLE
Added quaternionOffset and flipXYZ to default.vrsettings

### PR DESCRIPTION
**Pull request by lucas-vrtech**

### CODE REVIEW

File: LucidGloves/src/DeviceProvider.cpp [hunk 1]
```diff
@@ -29,8 +29,20 @@ VRDeviceConfiguration_t DeviceProvider::GetConfiguration(vr::ETrackedControllerR
...
+	const bool leftReversedX = vr::VRSettings()->GetBool(c_settingsSection, "leftFlippedX");
+	const bool leftReversedY = vr::VRSettings()->GetBool(c_settingsSection, "leftFlippedX");
+	const bool leftReversedZ = vr::VRSettings()->GetBool(c_settingsSection, "leftFlippedX");
+
+	//x axis may be flipped for the different hands
+	const vr::HmdVector3_t offsetVector = { role == vr::TrackedControllerRole_RightHand || !leftReversedX ? offsetX : -offsetX, offsetY, offsetZ };
+	const vr::HmdVector3_t angleOffsetVector =
+	  { role == vr::TrackedControllerRole_RightHand || !leftReversedX ? offsetXDeg : -offsetXDeg, 
+		role == vr::TrackedControllerRole_RightHand || !leftReversedY ? offsetYDeg : -offsetYDeg,
+		role == vr::TrackedControllerRole_RightHand || !leftReversedZ ? offsetZDeg : -offsetZDeg };
```
Comment (danwillm) [inline LucidGloves/src/DeviceProvider.cpp:45]: nitpick, but having a single `bool` for right handed-ness might make the logic easier to follow, as well as some brackets.

File: LucidGloves/include/Quaternion.h [hunk 2]
```diff
@@ -15,5 +15,6 @@ vr::HmdVector3_t GetPosition(vr::HmdMatrix34_t matrix);
 vr::HmdQuaternion_t MultiplyQuaternion(const vr::HmdQuaternion_t q, const vr::HmdQuaternion_t r);
 
 vr::HmdQuaternion_t QuaternionFromAngle(const double& xx, const double& yy, const double& zz, const double& a);
+vr::HmdQuaternion_t EulerToQuaternion(const double& xx, const double& yy, const double& zz);
```
Comment (danwillm) [inline LucidGloves/include/Quaternion.h:18]: Can these be the same argument/parameter names as what's provided in the definition?

File: LucidGloves/lucidgloves/resources/settings/default.vrsettings [hunk 3]
```diff
@@ -1,12 +1,19 @@
...
 		"y_offset": -0.08,
 		"z_offset": 0.07,
-		"pose_offset": -0.06
+		"pose_offset": -0.06,
+		"x_offsetDegrees": 90,
+		"y_offsetDegrees": 0,
+		"z_offsetDegrees": 0,
+		"leftFlippedX": true,
+		"leftFlippedY": false,
+		"leftFlippedZ": true
```
Comment (danwillm) [inline LucidGloves/lucidgloves/resources/settings/default.vrsettings:16]: Can these have the casing format as the other properties here?
Happy to switch to camelCase if you want.

File: LucidGloves/src/ControllerPose.cpp [hunk 4]
```diff
@@ -32,10 +35,13 @@ vr::DriverPose_t ControllerPose::UpdatePose() {
 			newPose.vecPosition[1] = trackedDevicePoses[m_shadowControllerId].mDeviceToAbsoluteTracking.m[1][3] + vectorOffset.v[1];
 			newPose.vecPosition[2] = trackedDevicePoses[m_shadowControllerId].mDeviceToAbsoluteTracking.m[2][3] + vectorOffset.v[2]; //- forward
 
-			vr::HmdQuaternion_t offset_quaternion = QuaternionFromAngle(1, 0, 0, DegToRad(-45));
-
+			//vr::HmdQuaternion_t offset_quaternion = QuaternionFromAngle(1, 0, 0, DegToRad(-45));
+			bool isRightHand = m_shadowDeviceOfRole == vr::TrackedControllerRole_RightHand;
```
Comment (danwillm) [inline LucidGloves/src/ControllerPose.cpp:39]: This won't be changing - perhaps best to add as a member variable?

File: LucidGloves/src/ControllerPose.cpp [hunk 5]
```diff
@@ -32,10 +35,13 @@ vr::DriverPose_t ControllerPose::UpdatePose() {
...
+			//vr::HmdQuaternion_t leftOffset_quaternion = { 0.707f, 0.00f, 0.00f, -0.707f };
+			vr::HmdQuaternion_t rightOffset_quaternion = { 0.707f, 0.00f, 0.00f, 0.707f };
```
Comment (danwillm) [inline LucidGloves/src/ControllerPose.cpp:41]: what's this for?

**Review by danwillm — CHANGES_REQUESTED** [review]

File: LucidGloves/lucidgloves/resources/settings/default.vrsettings (hunk 3 above)
Comment (lucas-vrtech) [reply LucidGloves/lucidgloves/resources/settings/default.vrsettings:16]: Oops, lol, will switch to underscores

File: LucidGloves/src/ControllerPose.cpp (hunk 5 above)
Comment (lucas-vrtech) [reply LucidGloves/src/ControllerPose.cpp:41]: removed

File: LucidGloves/src/ControllerPose.cpp (hunk 4 above)
Comment (lucas-vrtech) [reply LucidGloves/src/ControllerPose.cpp:39]: removed

File: LucidGloves/lucidgloves/resources/settings/default.vrsettings [hunk 6]
```diff
@@ -11,9 +11,9 @@
 		"x_offsetDegrees": 90,
 		"y_offsetDegrees": 0,
 		"z_offsetDegrees": 0,
```
Comment (danwillm) [inline LucidGloves/lucidgloves/resources/settings/default.vrsettings]: would be great for these also to be `snake_case` if you that style.

**Review by danwillm — CHANGES_REQUESTED** [review]
one more teeny change, then lgtm!

**Review by danwillm — APPROVED** [review]
LGTM!

**Pull request merged**